Organize report admin actions into single control

zarino opened the pull request:
Fixes #89.

Adds a new segmented control, containing the admin actions the current user is permitted to perform (currently any number of the 3 actions: moderation, inspection, and shortlisting).

![screen shot 2016-09-20 at 11 24 21](https://cloud.githubusercontent.com/assets/739624/18666650/dd0177aa-7f24-11e6-8dbc-700b6b129515.png)

Tested in IE8-10, and modern browsers. And screenreaders.

dracos commented on web/cobrands/fixmystreet/fixmystreet.js:
```diff
@@ -281,16 +281,25 @@ $.extend(fixmystreet.set_up, {
+                buttonLabel = $submit.data('label-add');
+                buttonValue = $submit.data('value-add');
             }
+            $change.val(changeVal);
```
This should be changeValue

zarino replied:
Done!

dracos commented on templates/web/base/report/_main.html:
```diff
@@ -8,19 +8,25 @@
+        data-value-add="[% loc('Shortlist') %]"
     [%~ IF c.user.is_planned_report(problem) ~%]
-        [% loc('Remove from planned reports') %]
+        value="[% loc('Shortlisted') %]"
```
Should shortlisted look pushed in, like the radio segmented controls do, perhaps?

dracos replied:
Then also Inspect could perhaps stay visible, also pushed in, when the inspect part is being shown, and clicking it could come out of inspect mode? Though might worry about losing info then I guess.

zarino replied:
> Should shortlisted look pushed in, like the radio segmented controls do, perhaps?
  
Ooh that’s an interesting idea!

> Then also Inspect could perhaps stay visible, also pushed in, when the inspect part is being shown, and clicking it could come out of inspect mode? Though might worry about losing info then I guess.

We already have a "Cancel" button in the Inspection form, that just links back to `/report/:id`, losing any changes you’ve made in the form. Maybe a pressed-in button labelled "Inspect" is a little more dangerous, because it isn't labelled with a danger word.

dracos replied:
Yeah, agreed; I still think it would look nice to be there, pushed in, but clicking it do nothing perhap?

zarino replied:
I guess the text could change on the pushed-in buttons – so pushing the "Shortlist" button turns it into a pushed-in, yellow "Shortlisted"; and pushing the "Inspect" button turns it into a pushed-in, yellow "Cancel" button.

Danger here is that, while "Shortlisted" makes sense as a label (since it's describing the current state of the stateful button – it's just a linguistic equivalent to the yellow, pushed-in styling), "Cancel" is an action rather than a state, so it doesn't match. The matching label for that button, I guess, would be "Inspecting", since that describes the state the report is currently in.

Bah.

zarino replied:
Also, this is the point at which I ask: Why do we have an "Inspect" button at all? Why doesn't the "Inspector" sidebar *always* display if you have the `report_inspect` permission?

@BenJam @davea ^

zarino commented:
@dracos – I've fixed the test to expect "Shortlist"/"Shortlisted" in place of the old button text, and also updated all the user-facing bits of UI to use the "Shortlist" terminology instead of "Planned reports".

The URLs, database fields, etc still use "`planned`" but I'm guessing it's safest to leave them as-is.

zarino commented:
@dracos – I've just pushed a commit that refactors `.radio-segmented-control` into `.segmented-control--radio`, a special-case of `.segmented-control`. Makes the styling a bit simpler too.
